pegasus/sites.v3/code.org/views/homepage_hero.haml

```diff
@@ -1,41 +1,33 @@
-- census_announcement = DCDO.get('census_announcement', nil)
-
--# Whether we show non-rotating background image and center options on desktop.
-- show_census_layout = request.language == "en"
-
--# Whether we have a special announcement to also show.
-- show_census_announcement = census_announcement && census_announcement['homepage_text'] && census_announcement['homepage_subtext'] && request.language == "en"
-
-- census_announcement_hero = [{"id":99,"id_s":"hero_census_announcement","type_s":"widehero","text1_s":"homepage_hero_text_stat_students","param1_s":"50% 30%","param2_s":"stat","param3_s":"bottom","param4_s":nil,"param5_s":nil,"watermark_s":nil,"image_t":"/images/homepage/announcement.jpg","site_s":"both","active_s":"show","language_supported_s":"all"}]
-
-- actions = DB[:cdo_homepage].where(type_s: "action", active_s: ["show","hide"], site_s: [request.site,"both"], language_supported_s: ["all", languageFilter]).all
+:ruby
```

Comment:
Can this be extracted to a standalone .rb file as well? 😄

Reply:
I'm not sure the best way to do this, since we don't really have a generic system for per-page controllers in Pegasus afaict.  Though maybe there's a good example that I'm not thinking of right now...

Reply:
Merging now to ahead of some other changes but still happy to brainstorm ways to do this well.